app/src/main/java/com/example/superimagesviewer/MosaicFragment.kt

```diff
@@ -32,7 +32,12 @@ class MosaicFragment : Fragment() {
         mosaicViewModel = ViewModelProvider(this).get(MosaicViewModel::class.java)
         initRecyclerView()
     }
-    
+
+    override fun onStop() {
+        super.onStop()
+        MosaicRepository.graphRequest.executeAsync().cancel(true)
```

Comment:
I think you did wrong way here. 
You want to have next result:
If application onStop() was called -> you cancel current request to graphAPI (findImagesByIds)
But what is really going on here is:
If application onStop() was called -> you start new async request with _executeAsync()_ method and in the same moment cancel it.
But old one still running..

Reply:
In order to fix this you need to save into variable result of graphRequest.executeAsync()
I mean something like this:
var findImagesByIdsAsyncTask = graphRequest.executeAsync()

So then you will be able do like this:

onStop() {
MosaicRepo.findImagesByIdsAsyncTask.Cancel()
}

Reply:
done